js/src/annotations/annotationTooltip.js

```diff
@@ -82,22 +81,11 @@
             jQuery(selector + ' a.cancel').on("click", function(event) {
               event.preventDefault();
 
-              var returnToPointer = function(){
-                _this.eventEmitter.publish('SET_STATE_MACHINE_POINTER.' + _this.windowId);
+              var cancelCallback = function(){
                 api.destroy();
-                if (params.onCancel) { params.onCancel(); }
               };
 
-              if (_this.activeEditor.isDirty()) {
-                new $.DialogBuilder().confirm(i18n.t('cancelAnnotation'),function(result){
```

Comment:
Looks like you are not passing the isDirty check in the event publisher so it's showing the dialog on an empty text editor.